compaction/compaction_manager.cc

```diff
@@ -802,7 +802,6 @@ future<> compaction_manager::rewrite_sstables(replica::table* t, sstables::compa
             auto sstable_level = sst->get_sstable_level();
             auto run_identifier = sst->run_identifier();
             auto sstable_set_snapshot = can_purge ? std::make_optional(t.get_sstable_set()) : std::nullopt;
-            // FIXME: this compaction should run with maintenance priority.
```

Comment:
I don't know if running rewrite_sstables() in maintenance group is a good idea.

That's because sstables locked by it influence compaction shares.

If compaction shares are increasing significantly, as new sstables arrive, we'd like to be able to speed up cleanup to make sure backlog can be decreased. Making this change is decoupling the controller from any maintenance operation.

Reply:
We can separate this change to let this PR be merged earlier as a first step and examine running rewrite_sstables in maintenance sg as a follow-up, maybe with additional improvements to address any issues caused by it.

For example, we could implement a controller for maintenance scheduling group shares that could take the backlog into account as well as other inputs.  In essence we should decouple maintenance scheduling from regular compaction scheduling.  They are serving different purposes.

As for cleanup to complete faster due to backlog, new sstables should not need cleanup as they are written after the respective topology change(s) are already done, so they shouldn't contain any tokens not belonging to the node. The reason to speed up cleanup to reduce the backlog is that it prevents regular compaction from selecting sstables before they are cleaned up.

If we had a `requires_cleanup` flag on the sstable we could mark them when selected for cleanup at the beginning of the cleanup task but not register them as compacting, so that regular compaction can select them too and apply the cleanup filter when reading the sstable as part of regular compaction. When cleanup gets to processing the sstables, it can register them one at a time, as they are processed, to prevent regular compaction from accessing them concurrently.  If the sstable was already compacted and is now unlinked, cleanup can just skip it.